components/file_preview/file_preview.tsx

```diff
@@ -3,24 +3,20 @@
 
 import React, {ReactNode} from 'react';
 import {getFileThumbnailUrl, getFileUrl} from 'mattermost-redux/utils/file_utils';
+import {FileInfo} from 'mattermost-redux/types/files';
 
-import FilenameOverlay from 'components/file_attachment/filename_overlay.jsx';
+import FilenameOverlay from 'components/file_attachment/filename_overlay';
 import Constants, {FileTypes} from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
 import FileProgressPreview from './file_progress_preview';
 
-export type FilePreviewInfo = {
-    extension?: string;
-    id: string;
-    width?: number;
-    height?: number;
-    has_preview_image?: boolean;
-    size?: number;
-    create_at?: string;
+type UploadInfo = {
+    name: string;
     percent?: number;
     type?: string;
 }
+export type FilePreviewInfo = FileInfo & UploadInfo;
```

Comment:
This is pricipal change in ```file_preview``` module. This is required as the FilePreviewInfo object is used both as FileInfo and UploadInfo. This way the resulting type has both the types required by FileInfo and UploadInfo.